src/librustc/ty/error.rs

```diff
@@ -58,6 +58,8 @@ pub enum TypeError<'tcx> {
     ConstMismatch(ExpectedFound<&'tcx ty::Const<'tcx>>),
 
     IntrinsicCast,
+    /// Safe `#[target_feature]` functions are not assignable to safe function pointers.
+    TargetFeatureCast,
```

Comment:
Not sure if it belongs here, or should be part of `TypeError::Mismatch`

Reply:
It seems fine to me like this.